From: Ivan Kokshaysky <ink@jurassic.park.msu.ru>
Date: Sun, 30 Oct 2005 01:15:43 +0000 (-0700)
Subject:     [PATCH] fix alpha breakage
X-Git-Url: http://www.kernel.org/git/gitweb.cgi?p=linux/kernel/git/torvalds/linux-2.6.git;a=commitdiff;h=63f324cf0792ed69089b79d6921ba3aaea97af50

  [PATCH] fix alpha breakage
  
  barrier.h uses barrier() in non-SMP case.  And doesn't include compiler.h.
  
  Cc: Al Viro <viro@ftp.linux.org.uk>
  Signed-off-by: Andrew Morton <akpm@osdl.org>
  Signed-off-by: Linus Torvalds <torvalds@osdl.org>
---

--- a/include/asm-alpha/barrier.h
+++ b/include/asm-alpha/barrier.h
@@ -1,6 +1,8 @@
 #ifndef __BARRIER_H
 #define __BARRIER_H
 
+#include <asm/compiler.h>
+
 #define mb() \
 __asm__ __volatile__("mb": : :"memory")
 
